bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/discovery/LutronMdnsBridgeDiscoveryService.java

```diff
@@ -0,0 +1,184 @@
+@NonNullByDefault
+public class LutronMdnsBridgeDiscoveryService implements MDNSDiscoveryParticipant {
+
+    // Lutron mDNS service <app>.<protocol>.<servicedomain>
+    private static final String LUTRON_MDNS_SERVICE_TYPE = "_lutron._tcp.local.";
+
+    private static final String PRODFAM_CASETA = "Caseta";
+    private static final String PRODTYP_CASETA_SBP2 = "Smart Bridge Pro 2";
+    private static final String DEVCLASS_CASETA_SBP2 = "08050100";
+    private static final String DEFAULT_LABEL = "Unknown Lutron bridge";
+
+    private static final Pattern HOSTNAME_REGEX = Pattern.compile("lutron-([0-9a-f]+)\\."); // ex: lutron-01f1529a.local
+
+    private final Logger logger = LoggerFactory.getLogger(LutronMdnsBridgeDiscoveryService.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(THING_TYPE_IPBRIDGE);
+    }
+
+    @Override
+    public String getServiceType() {
+        return LUTRON_MDNS_SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (!service.hasData()) {
+            return null;
+        }
+
+        String nice = service.getNiceTextString();
+        String qualifiedName = service.getQualifiedName();
+
+        InetAddress[] ipAddresses = service.getInetAddresses();
+        String devclass = service.getPropertyString("DEVCLASS");
+        String codever = service.getPropertyString("CODEVER");
+        String macaddr = service.getPropertyString("MACADDR");
+
+        logger.debug("Lutron mDNS bridge discovery notified of Lutron mDNS service: {}", nice);
+        logger.trace("Lutron mDNS service qualifiedName: {}", qualifiedName);
+        logger.trace("Lutron mDNS service ipAddresses: {} ({})", ipAddresses, ipAddresses.length);
+        logger.trace("Lutron mDNS service property DEVCLASS: {}", devclass);
+        logger.trace("Lutron mDNS service property CODEVER: {}", codever);
+        logger.trace("Lutron mDNS service property MACADDR: {}", macaddr);
+
+        Map<String, Object> properties = new HashMap<>();
+        String label = DEFAULT_LABEL;
+
+        if (ipAddresses.length < 1) {
+            return null;
+        }
+        if (ipAddresses.length > 1) {
+            logger.debug("Multiple addresses found for discovered Lutron device. Using only the first.");
+        }
+        properties.put(HOST, ipAddresses[0].getHostAddress());
+
+        String bridgeHostName = ipAddresses[0].getHostName();
+        logger.debug("Lutron mDNS bridge hostname: {}", bridgeHostName);
+
+        if (devclass != null && devclass.equals(DEVCLASS_CASETA_SBP2)) {
+            properties.put(PROPERTY_PRODFAM, PRODFAM_CASETA);
+            properties.put(PROPERTY_PRODTYP, PRODTYP_CASETA_SBP2);
+            label = PRODFAM_CASETA + " " + PRODTYP_CASETA_SBP2;
+        } else {
+            logger.info("Lutron device with unknown DEVCLASS discovered via mDNS: {}. Configure device manually.",
```

Comment:
I think we can reduce logging level to DEBUG here. Otherwise it may spam log files if a user is not interested in discovery or does not have the chance to solve it.

Reply:
Well, I purposely set this to INFO here so that users would be encouraged to report unknown DEVCLASS numbers. The one for Caseta is the most common, but I've had no luck in finding (or getting people to report) the numbers for less common devices.  Once users report them, I can add support for them very quickly.